test/unit/events.js

```diff
@@ -125,4 +125,55 @@ describe('Backbone.Eventrouter Events', function() {
       });
     });
   });
+
+  describe('when a router is destroyed', function() {
+    describe('when a route is removed', function() {
+      beforeEach(function() {
+        this.beforeRouteStub = this.sinon.stub();
+        this.routeStub = this.sinon.stub();
+        this.beforeRouteNameStub = this.sinon.stub();
+        this.routeNameStub = this.sinon.stub();
+
+        this.myEventRouter.on('before:route', this.beforeRouteStub);
+        this.myEventRouter.on('route', this.routeStub);
+        this.myEventRouter.on('before:route:testevent:bar', this.beforeRouteNameStub);
+        this.myEventRouter.on('route:testevent:bar', this.routeNameStub);
+
+        this.myEventRouter.addRouteTrigger('test/url/:foo', 'testevent:bar');
+        this.myEventRouter.destroy();
+        this.myEventRouter.navigate('test/url/baz', { trigger: true });
+      });
+
+      it('should not throw a before:route event', function() {
+        expect(this.beforeRouteStub).to.not.have.been.calledOnce;
+      });
+    });
+
+    describe('when a route is removed and added to a new router', function() {
+      beforeEach(function() {
+        this.beforeRouteStub = this.sinon.stub();
+        this.routeStub = this.sinon.stub();
+        this.beforeRouteNameStub = this.sinon.stub();
+        this.routeNameStub = this.sinon.stub();
+
+        this.myEventRouter.on('before:route', this.beforeRouteStub);
+        this.myEventRouter.on('route', this.routeStub);
+        this.myEventRouter.on('before:route:testevent:bar', this.beforeRouteNameStub);
+        this.myEventRouter.on('route:testevent:bar', this.routeNameStub);
+
+        this.myEventRouter.addRouteTrigger('test/url/:foo', 'testevent:bar');
+        this.myEventRouter.destroy();
+
+        this.myOtherEventRouter = new Backbone.EventRouter();
+        this.myOtherEventRouter.getChannel();
+
+        this.myOtherEventRouter.addRouteTrigger('test/url/:foo', 'testevent:bar');
+        this.myOtherEventRouter.navigate('test/url/baz', { trigger: true });
+      });
+
+      it('should throw a before:route event', function() {
+        expect(this.beforeRouteStub).to.have.been.calledOnce;
+      });
```

Comment:
I don't get why this test doesn't pass.

I'm trying to do the following:

1. Add a route to the event router and then destroy it.
2. Create a new router and add the same route to it.
3. Go to that route and verify the event triggers.

Reply:
The `it`s should be reformatted, but essentially needed a stub on the second router and needed to route to different urls

```js
    describe('when a route is removed and added to a new router', function() {
      beforeEach(function() {
        this.beforeRouteStub = this.sinon.stub();
        this.beforeRouteOtherStub = this.sinon.stub();

        this.myEventRouter.on('before:route', this.beforeRouteStub);

        this.myEventRouter.addRouteTrigger('test/url/:foo', 'testevent:bar');
        this.myEventRouter.navigate('test/url/baz', { trigger: true });
        this.myEventRouter.destroy();
        this.myEventRouter.navigate('test/url/baz2', { trigger: true });

        this.myOtherEventRouter = new Backbone.EventRouter();
        this.myOtherEventRouter.on('before:route', this.beforeRouteOtherStub);
        this.myOtherEventRouter.addRouteTrigger('test/url/:foo', 'testevent:bar');
        this.myOtherEventRouter.navigate('test/url/baz3', { trigger: true });
      });

      it('should throw a before:route event', function() {
        expect(this.beforeRouteStub).to.have.been.calledOnce;
      });

      it('should throw a before:route event', function() {
        expect(this.beforeRouteOtherStub).to.have.been.calledOnce;
      });
    });
```

Reply:
Just updated this test.